concordium-consensus/tests/scheduler/SchedulerTests/Helpers.hs

```diff
@@ -1,24 +1,430 @@
-import Data.Word
+import Concordium.TimeMonad
 
-getResults :: [(a, TransactionSummary)] -> [(a, ValidResult)]
-getResults = map (\(x, r) -> (x, tsResult r))
+getResults :: [(a, Types.TransactionSummary)] -> [(a, Types.ValidResult)]
+getResults = map (\(x, r) -> (x, Types.tsResult r))
 
 -- | The cost for processing a simple transfer (account to account)
 -- with one signature in the transaction.
 --
 -- * @SPEC: <$DOCS/Transactions#transaction-cost-header-simple-transfer>
-simpleTransferCost :: Energy
+simpleTransferCost :: Types.Energy
 simpleTransferCost = Cost.baseCost (Types.transactionHeaderSize + 41) 1 + Cost.simpleTransferCost
 
-simpleTransferCostWithMemo1 :: Word64 -> Energy
-simpleTransferCostWithMemo1 memoSize = Cost.baseCost (Types.transactionHeaderSize + 41 + 2 + memoSize) 1
+simpleTransferCostWithMemo1 :: Word64 -> Types.Energy
+simpleTransferCostWithMemo1 memoSize =
+    Cost.baseCost
+        (Types.transactionHeaderSize + 41 + 2 + memoSize)
+        1
+
+simpleTransferCostWithMemo2 :: Word64 -> Types.Energy
+simpleTransferCostWithMemo2 memoSize =
+    Cost.baseCost
+        (Types.transactionHeaderSize + 41 + 2 + memoSize)
+        1
+        + Cost.simpleTransferCost
+
+-- | Monad that implements the necessary constraints to be used for running the scheduler.
+newtype PersistentBSM pv a = PersistentBSM
+    { _runPersistentBSM ::
+        BS.PersistentBlockStateMonad
+            pv
+            (BS.PersistentBlockStateContext pv)
+            (Blob.BlobStoreM' (BS.PersistentBlockStateContext pv))
+            a
+    }
+    deriving
+        ( Applicative,
+          Functor,
+          Monad,
+          BS.AccountOperations,
+          BS.ContractStateOperations,
+          BS.ModuleQuery,
+          MonadProtocolVersion,
+          BlockStateTypes,
+          Blob.MonadBlobStore,
+          BS.BlockStateQuery,
+          BS.BlockStateOperations,
+          MonadIO,
+          MonadCache BS.ModuleCache,
+          BS.BlockStateStorage
+        )
+
+deriving instance
+    (Types.AccountVersionFor pv ~ av) =>
+    MonadCache (BS.AccountCache av) (PersistentBSM pv)
+
+instance MonadLogger (PersistentBSM pv) where
+    logEvent _ _ _ = return ()
+
+instance TimeMonad (PersistentBSM pv) where
+    currentTime = return $ read "1970-01-01 13:27:13.257285424 UTC"
+
+-- |Call a function for each protocol version, returning a list of results.
+-- This is used to run a test against every protocol version.
+forEveryProtocolVersion ::
+    (forall pv. (Types.IsProtocolVersion pv) => Types.SProtocolVersion pv -> String -> a) ->
+    [a]
+forEveryProtocolVersion check =
+    [ check Types.SP1 "P1",
+      check Types.SP2 "P2",
+      check Types.SP3 "P3",
+      check Types.SP4 "P4",
+      check Types.SP5 "P5"
+    ]
+
+-- |Construct a test block state containing the provided accounts.
+createTestBlockStateWithAccounts ::
+    (Types.IsProtocolVersion pv) =>
+    [Basic.Account (Types.AccountVersionFor pv)] ->
```

Comment:
Why use `basic` accounts here. WHy not have a helper for constructing persistent ones, and take those in here?

Reply:
`BS.initialPersistentState` uses `intialState` from Basic, which requires some work to port to persistent state, so I'll address this in another PR.